src/ChartWidget/IncompleteData/utils.ts

```diff
@@ -22,3 +25,32 @@ export function checkShouldShowBanner(): boolean {
   const value = getSavedValue(INCOMPLETE_DATA_BANNER_CLOSE_DATE)
   return !value || +value < Date.now()
 }
+
+export type TrackUpgrageProps = {
+  e: MouseEvent
+  restrictedMetrics: any[]
+  isLoggedIn: boolean
+  location: string
+}
+
+export function trackUpgrade({ e, restrictedMetrics, isLoggedIn, location }: TrackUpgrageProps) {
```

Comment:
Name of the function implies that it's used only for analytic's purposes. Better to rename it to `onUpgradeClick`